Migrate existing print statements to use Go logging library

# Summary

Migrate existing print statements to use Go's standard library for logging


# Background

**Is your improvement related to a problem? Please describe**:

Currently the only logging we use in the Go port are print statements.

**Describe the solution you'd like**:

Per #213, we should use the Go standard library for logging. This will give us more flexibility in verbosity levels and formatting later on.

**Describe alternatives you've considered**:

We considered `logrus` but decided to keep it simple and use the standard library to keep our dependencies down.


# Details

This is where I could find any existing use of print statements in our code:

```bash
$ grep --exclude-dir={docs/,.git/} -rnw . -i -e '.*print.*'

./cmd/teleirc.go:29:		fmt.Printf("Current TeleIRC version: %s\n", version)
./cmd/teleirc.go:35:		fmt.Printf("Debug mode currently set to: %t\n", *flagDebug)
./cmd/teleirc.go:40:		fmt.Fprintln(os.Stderr, err)
./cmd/teleirc.go:55:		fmt.Println(ircErr)
./cmd/teleirc.go:57:		fmt.Println(tgErr)
./internal/handlers/telegram/telegram.go:25:	fmt.Println("Creating new Telegram bot client...")
./internal/handlers/telegram/telegram.go:43:	fmt.Println("Starting up Telegram bot...")
./internal/handlers/telegram/telegram.go:47:		fmt.Println("Failed to connect to Telegram")
./internal/handlers/irc/handlers.go:47:		c.sendToTg(fmt.Sprintf(joinFmt, e.Source.Name))
./internal/handlers/irc/handlers.go:53:		c.sendToTg(fmt.Sprintf(partFmt, e.Source.Name))
./internal/handlers/irc/handlers.go:59:		c.sendToTg(fmt.Sprintf(quitFmt, e.Source.Name, e.Params[0]))
./internal/handlers/irc/irc.go:24:	fmt.Println("Creating new IRC bot client...")
./internal/handlers/irc/irc.go:45:	fmt.Println("Starting up IRC bot...")
./internal/handlers/irc/irc.go:68:	fmt.Println("Adding IRC event handlers...")
```


# Outcome

Consistent use of the Go library for logging where we are currently using print statements

The `log` package is thread safe (yay goroutines). I think behind the scenes `log` writes a string via `os.pipe` into an virtual file/iowriter [`os.Stderr`, by default](https://golang.org/pkg/os/#pkg-variables).
I made a test go project `randomLogDemo`, to see how log writes into `os.Stdout` and `os.Stderr`. We cant use `logger.Panicln` or `logger.Fataln` because it would terminate the process regardless if it is writing to `os.Stdout`. Furthermore, not every log/print message is an error, thus `os.Stderr` would not be appropriate for ever log message type; we would need to seperate log messages into different topics of concerns. From the [DIY logging framework article](https://www.ardanlabs.com/blog/2013/11/using-log-package-in-go.html) during my research, we could create a `logger` object for each topic - `Info`, `Warning`, `Error`. I know we discuss a start-up flag (`-v`) to declare verbosity, this would be the foundation for that effort.

Let me know what you guys think. Once we come to agreement, @10eMyrT and I can start implimenting these and replace the above `fmt` locations, and we can all share the same logging practice

Proposed design:
```go
package main

import (
	"io/ioutil"
	"log"
	"os"
)

var (
	Info    *log.Logger
	Warning *log.Logger
	Error   *log.Logger
	Ignored *log.Logger
)

func main() {
	logFlags := log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
	Info = log.New(os.Stdout, "INFO: ", logFlags)
	Warning = log.New(os.Stdout, "WARNING: ", logFlags)
	Error = log.New(os.Stderr, "ERROR: ", logFlags)
	Ignored = log.New(ioutil.Discard, "ERROR: ", logFlags)

	Info.Println("Pushing string to Info")
	Warning.Println("Pushing string to Warning")
	Ignored.Println("Pushing string to Ignored")

	filename := "fakefile.txt"
	Info.Println("Attepting to Open", filename)
	_, err := os.Open(filename)
	if err != nil {
		Error.Println(err)
	} else {
		Info.Println("Reading", filename)
	}
}
```

```
INFO: 2020/02/23 18:13:35.202738 C:/Users/Kennedy/Documents/projects/randomLogDemo/logmenow.go:23: Pushing string to Info
WARNING: 2020/02/23 18:13:35.202738 C:/Users/Kennedy/Documents/projects/randomLogDemo/logmenow.go:24: Pushing string to Warning
INFO: 2020/02/23 18:13:35.202738 C:/Users/Kennedy/Documents/projects/randomLogDemo/logmenow.go:27: Attepting to Open fakefile.txt
ERROR: 2020/02/23 18:13:35.202738 C:/Users/Kennedy/Documents/projects/randomLogDemo/logmenow.go:31: open fakefile.txt: The system cannot find the file specified.
```

```go
package main

import (
	"log"
	"os"
)

func main() {
	logFlags := log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
	logger := log.New(os.Stdout, "", logFlags)
	logger.Panicln("Pushing string to Panicln") // Terminates and dumps memory trace
	logger.Fatalln("Pushing string to Fatalln") // Terminates
	logger.Println("Pushing string to Println")
}
```
```
2020/02/23 17:47:31.893368 C:/Users/Kennedy/Documents/projects/randomLogDemo/logmenow.go:11: Pushing string to Panicln
panic: Pushing string to Panicln


goroutine 1 [running]:
log.(*Logger).Panicln(0xc0000a8050, 0xc0000a5f40, 0x1, 0x1)
	C:/Go/src/log/log.go:226 +0xb1
main.main()
	C:/Users/Kennedy/Documents/projects/randomLogDemo/logmenow.go:11 +0xd9
```

--- a/cmd/teleirc.go
+++ b/cmd/teleirc.go
@@ -3,8 +3,9 @@ package main
 
 import (
 	"flag"
-	"fmt"
 	"os"
+	"log"
+	"io/ioutil"
 
 	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api"
 	"github.com/ritlug/teleirc/internal"
@@ -20,24 +21,30 @@ var (
 	flagPath    = flag.String("conf", ".env", "config file")
 	flagDebug   = flag.Bool("debug", false, "enable debugging")
 	flagVersion = flag.Bool("version", false, "displays current version of TeleIRC")
+
+	logFlags    = log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
+	Info        = log.New(os.Stdout, "INFO: ", logFlags)
+	Warning     = log.New(os.Stdout, "WARNING: ", logFlags)
+	Error       = log.New(os.Stderr, "ERROR: ", logFlags)
+	Ignored     = log.New(ioutil.Discard, "ERROR: ", logFlags)
 )
 
 func main() {
 	flag.Parse()
 
 	if *flagVersion {
-		fmt.Printf("Current TeleIRC version: %s\n", version)
+		Info.Printf("Current TeleIRC version: %s\n", version)
 		return
 	}
 
 	// TODO: Build out debugging capabilities for more verbose output
 	if *flagDebug {
-		fmt.Printf("Debug mode currently set to: %t\n", *flagDebug)
+		Info.Printf("Debug mode currently set to: %t\n", *flagDebug)
 	}
 
 	settings, err := internal.LoadConfig(*flagPath)
 	if err != nil {
-		fmt.Fprintln(os.Stderr, err)
+		Error.Println(err)
 		os.Exit(1)
 	}
 
@@ -52,8 +59,8 @@ func main() {
 
 	select {
 	case ircErr := <-ircChan:
-		fmt.Println(ircErr)
+		Error.Println(ircErr)
 	case tgErr := <-tgChan:
-		fmt.Println(tgErr)
+		Error.Println(tgErr)
 	}
 }

--- a/cmd/teleirc.go
+++ b/cmd/teleirc.go
@@ -24,7 +24,7 @@ var (
 func main() {
 	flag.Parse()
 
-	verbose := internal.Debug {DebugLevel:  *flagDebug	}
+	verbose := internal.Debug { DebugLevel:  *flagDebug }
 
 	if *flagVersion {
 		verbose.PrintVersion("Current TeleIRC version: " + version)

--- a/cmd/teleirc.go
+++ b/cmd/teleirc.go
@@ -27,7 +27,7 @@ func main() {
 	verbose := internal.Debug { *flagDebug	}
 
 	if *flagVersion {
-		verbose.LogInfo("Current TeleIRC version: " + version)
+		verbose.PrintVersion("Current TeleIRC version: " + version)
 		return
 	}
 

--- a/cmd/teleirc.go
+++ b/cmd/teleirc.go
@@ -24,9 +24,7 @@ var (
 
 	logFlags    = log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
 	Info        = log.New(os.Stdout, "INFO: ", logFlags)
-	Warning     = log.New(os.Stdout, "WARNING: ", logFlags)
 	Error       = log.New(os.Stderr, "ERROR: ", logFlags)
-	Ignored     = log.New(ioutil.Discard, "ERROR: ", logFlags)
 )
 
 func main() {

--- a/cmd/teleirc.go
+++ b/cmd/teleirc.go
@@ -24,7 +24,7 @@ var (
 func main() {
 	flag.Parse()
 
-	verbose := internal.Debug { *flagDebug	}
+	verbose := internal.Debug {DebugLevel:  *flagDebug	}
 
 	if *flagVersion {
 		verbose.PrintVersion("Current TeleIRC version: " + version)

--- a/cmd/teleirc.go
+++ b/cmd/teleirc.go
@@ -4,8 +4,6 @@ package main
 import (
 	"flag"
 	"os"
-	"log"
-	"io/ioutil"
 
 	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api"
 	"github.com/ritlug/teleirc/internal"
@@ -19,46 +17,43 @@ const (
 
 var (
 	flagPath    = flag.String("conf", ".env", "config file")
-	flagDebug   = flag.Bool("debug", false, "enable debugging")
+	flagDebug   = flag.Bool("debug", false, "disable debugging")
 	flagVersion = flag.Bool("version", false, "displays current version of TeleIRC")
-
-	logFlags    = log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
-	Info        = log.New(os.Stdout, "INFO: ", logFlags)
-	Error       = log.New(os.Stderr, "ERROR: ", logFlags)
 )
 
 func main() {
 	flag.Parse()
 
+	verbose := internal.Debug { *flagDebug	}
+
 	if *flagVersion {
-		Info.Printf("Current TeleIRC version: %s\n", version)
+		verbose.LogInfo("Current TeleIRC version: " + version)
 		return
 	}
 
 	// TODO: Build out debugging capabilities for more verbose output
-	if *flagDebug {
-		Info.Printf("Debug mode currently set to: %t\n", *flagDebug)
-	}
+	// Notify that debug is enabled
+	verbose.LogInfo("Debug mode enabled!")
 
 	settings, err := internal.LoadConfig(*flagPath)
 	if err != nil {
-		Error.Println(err)
+		verbose.LogError(err)
 		os.Exit(1)
 	}
 
 	var tgapi *tgbotapi.BotAPI
-	tgClient := tg.NewClient(settings.Telegram, tgapi)
+	tgClient := tg.NewClient(settings.Telegram, tgapi, verbose)
 	tgChan := make(chan error)
 
-	ircClient := irc.NewClient(settings.IRC)
+	ircClient := irc.NewClient(settings.IRC, verbose)
 	ircChan := make(chan error)
 	go ircClient.StartBot(ircChan, tgClient.SendMessage)
 	go tgClient.StartBot(tgChan, ircClient.SendMessage)
 
 	select {
 	case ircErr := <-ircChan:
-		Error.Println(ircErr)
+		verbose.LogError(ircErr)
 	case tgErr := <-tgChan:
-		Error.Println(tgErr)
+		verbose.LogError(tgErr)
 	}
 }

--- a/internal/debug.go
+++ b/internal/debug.go
@@ -36,3 +36,7 @@ func (d Debug) LogWarning(message string) {
 		warning.Println(message)
 	}
 }
+
+func (d Debug) PrintVersion(message string) {
+	info.Println(message)
+}

--- a/internal/debug.go
+++ b/internal/debug.go
@@ -17,6 +17,14 @@ var (
 	warning     = log.New(os.Stderr, "WARNING: ", logFlags)
 )
 
+
+type DebugLogger interface {
+	LogInfo(string)
+	LogError(error)
+	LogWarning(string)
+	PrintVersion(string)
+}
+
 type Debug struct {
 	DebugLevel  bool
 }

--- a/internal/debug.go
+++ b/internal/debug.go
@@ -1,0 +1,38 @@
+/*
+	Debug contains information about debug levels and define
+	debug statements
+*/
+package internal
+
+import (
+	"log"
+	"os"
+)
+
+var (
+	// logFlags    = log.Ldate | log.Ltime | log.Lshortfile
+	logFlags    = log.Ldate | log.Ltime
+	info = log.New(os.Stdout, "INFO: ", logFlags)
+	errorLog    = log.New(os.Stderr, "ERROR: ", logFlags)
+	warning     = log.New(os.Stderr, "WARNING: ", logFlags)
+)
+
+type Debug struct {
+	DebugLevel  bool
+}
+
+func (d Debug) LogInfo(message string) {
+	if d.DebugLevel {
+		info.Println(message)
+	}
+}
+
+func (d Debug) LogError(message error) {
+	errorLog.Println(message)
+}
+
+func (d Debug) LogWarning(message string) {
+	if d.DebugLevel {
+		warning.Println(message)
+	}
+}

--- a/internal/handlers/irc/irc.go
+++ b/internal/handlers/irc/irc.go
@@ -13,14 +13,14 @@ and the IRCSettings that were passed into NewClient
 type Client struct {
 	*girc.Client
 	Settings internal.IRCSettings
-	verbose internal.Debug
+	verbose internal.DebugLogger
 	sendToTg func(string)
 }
 
 /*
 NewClient returns a new IRCClient based on the provided settings
 */
-func NewClient(settings internal.IRCSettings, debug internal.Debug) Client {
+func NewClient(settings internal.IRCSettings, debug internal.DebugLogger) Client {
 	debug.LogInfo("Creating new IRC bot client...")
 	client := girc.New(girc.Config{
 		Server: settings.Server,

--- a/internal/handlers/irc/irc.go
+++ b/internal/handlers/irc/irc.go
@@ -1,8 +1,6 @@
 package irc
 
 import (
-	"log"
-	"io/ioutil"
 
 	"github.com/lrstanley/girc"
 	"github.com/ritlug/teleirc/internal"
@@ -15,23 +13,15 @@ and the IRCSettings that were passed into NewClient
 type Client struct {
 	*girc.Client
 	Settings internal.IRCSettings
+	verbose internal.Debug
 	sendToTg func(string)
 }
 
 /*
-Creating variables for logging
-*/
-var (
-	logFlags    = log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
-	Info        = log.New(os.Stdout, "INFO: ", logFlags)
-	Error       = log.New(os.Stderr, "ERROR: ", logFlags)
-)
-
-/*
 NewClient returns a new IRCClient based on the provided settings
 */
-func NewClient(settings internal.IRCSettings) Client {
-	Info.Println("Creating new IRC bot client...")
+func NewClient(settings internal.IRCSettings, debug internal.Debug) Client {
+	debug.LogInfo("Creating new IRC bot client...")
 	client := girc.New(girc.Config{
 		Server: settings.Server,
 		Port:   settings.Port,
@@ -44,7 +34,7 @@ func NewClient(settings internal.IRCSettings) Client {
 			Pass: settings.NickServPassword,
 		}
 	}
-	return Client{client, settings, nil}
+	return Client{client, settings, debug, nil}
 }
 
 /*
@@ -52,12 +42,12 @@ StartBot adds necessary handlers to the client and then connects,
 returns any errors that occur
 */
 func (c Client) StartBot(errChan chan<- error, sendMessage func(string)) {
-	Info.Println("Starting up IRC bot...")
+	c.verbose.LogInfo("Starting up IRC bot...")
 	c.sendToTg = sendMessage
 	c.addHandlers()
 	if err := c.Connect(); err != nil {
 		errChan <- err
-		Error.Println(err)
+		c.verbose.LogError(err)
 	} else {
 		errChan <- nil
 	}
@@ -76,7 +66,7 @@ addHandlers adds handlers for the client struct based on the settings
 that were passed in to NewClient
 */
 func (c Client) addHandlers() {
-	Info.Println("Adding IRC event handlers...")
+	c.verbose.LogInfo("Adding IRC event handlers...")
 	for eventType, handler := range getHandlerMapping() {
 		c.Handlers.Add(eventType, handler(c))
 	}

--- a/internal/handlers/irc/irc.go
+++ b/internal/handlers/irc/irc.go
@@ -24,9 +24,7 @@ Creating variables for logging
 var (
 	logFlags    = log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
 	Info        = log.New(os.Stdout, "INFO: ", logFlags)
-	Warning     = log.New(os.Stdout, "WARNING: ", logFlags)
 	Error       = log.New(os.Stderr, "ERROR: ", logFlags)
-	Ignored     = log.New(ioutil.Discard, "ERROR: ", logFlags)
 )
 
 /*
@@ -59,6 +57,7 @@ func (c Client) StartBot(errChan chan<- error, sendMessage func(string)) {
 	c.addHandlers()
 	if err := c.Connect(); err != nil {
 		errChan <- err
+		Error.Println(err)
 	} else {
 		errChan <- nil
 	}

--- a/internal/handlers/irc/irc.go
+++ b/internal/handlers/irc/irc.go
@@ -1,7 +1,8 @@
 package irc
 
 import (
-	"fmt"
+	"log"
+	"io/ioutil"
 
 	"github.com/lrstanley/girc"
 	"github.com/ritlug/teleirc/internal"
@@ -18,10 +19,21 @@ type Client struct {
 }
 
 /*
+Creating variables for logging
+*/
+var (
+	logFlags    = log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
+	Info        = log.New(os.Stdout, "INFO: ", logFlags)
+	Warning     = log.New(os.Stdout, "WARNING: ", logFlags)
+	Error       = log.New(os.Stderr, "ERROR: ", logFlags)
+	Ignored     = log.New(ioutil.Discard, "ERROR: ", logFlags)
+)
+
+/*
 NewClient returns a new IRCClient based on the provided settings
 */
 func NewClient(settings internal.IRCSettings) Client {
-	fmt.Println("Creating new IRC bot client...")
+	Info.Println("Creating new IRC bot client...")
 	client := girc.New(girc.Config{
 		Server: settings.Server,
 		Port:   settings.Port,
@@ -42,7 +54,7 @@ StartBot adds necessary handlers to the client and then connects,
 returns any errors that occur
 */
 func (c Client) StartBot(errChan chan<- error, sendMessage func(string)) {
-	fmt.Println("Starting up IRC bot...")
+	Info.Println("Starting up IRC bot...")
 	c.sendToTg = sendMessage
 	c.addHandlers()
 	if err := c.Connect(); err != nil {
@@ -65,7 +77,7 @@ addHandlers adds handlers for the client struct based on the settings
 that were passed in to NewClient
 */
 func (c Client) addHandlers() {
-	fmt.Println("Adding IRC event handlers...")
+	Info.Println("Adding IRC event handlers...")
 	for eventType, handler := range getHandlerMapping() {
 		c.Handlers.Add(eventType, handler(c))
 	}

--- a/internal/handlers/irc/irc_test.go
+++ b/internal/handlers/irc/irc_test.go
@@ -15,7 +15,10 @@ func TestNewClientBasic(t *testing.T) {
 		Port:    1234,
 		BotName: "test_name",
 	}
-	client := NewClient(ircSettings)
+	logger := internal.Debug{
+		DebugLevel:  false,
+	}
+	client := NewClient(ircSettings, logger)
 
 	expectedPing, _ := time.ParseDuration("20s")
 	expectedConfig := girc.Config{
@@ -36,7 +39,10 @@ func TestNewClientFull(t *testing.T) {
 		BotName:          "test_name",
 		NickServPassword: "test_pass",
 	}
-	client := NewClient(ircSettings)
+	logger := internal.Debug{
+		DebugLevel:  false,
+	}
+	client := NewClient(ircSettings, logger)
 	expectedPing, _ := time.ParseDuration("20s")
 	expectedConfig := girc.Config{
 		Server:    "test_server",

--- a/internal/handlers/telegram/telegram.go
+++ b/internal/handlers/telegram/telegram.go
@@ -2,8 +2,6 @@
 package telegram
 
 import (
-	"log"
-	"io/ioutil"
 
 	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api"
 	"github.com/ritlug/teleirc/internal"
@@ -16,24 +14,16 @@ the TelegramSettings needed to run the bot
 type Client struct {
 	api       *tgbotapi.BotAPI
 	Settings  internal.TelegramSettings
+	verbose   internal.Debug
 	sendToIrc func(string)
 }
 
 /*
-Creating variables for logging
-*/
-var (
-	logFlags    = log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
-	Info        = log.New(os.Stdout, "INFO: ", logFlags)
-	Error       = log.New(os.Stderr, "ERROR: ", logFlags)
-)
-
-/*
 NewClient creates a new Telegram bot client
 */
-func NewClient(settings internal.TelegramSettings, tgapi *tgbotapi.BotAPI) *Client {
-	Info.Println("Creating new Telegram bot client...")
-	return &Client{api: tgapi, Settings: settings}
+func NewClient(settings internal.TelegramSettings, tgapi *tgbotapi.BotAPI, debug internal.Debug) *Client {
+	debug.LogInfo("Creating new Telegram bot client...")
+	return &Client{api: tgapi, Settings: settings, verbose: debug}
 }
 
 /*
@@ -50,11 +40,12 @@ StartBot adds necessary handlers to the client and then connects,
 returning any errors that occur
 */
 func (tg *Client) StartBot(errChan chan<- error, sendMessage func(string)) {
-	Info.Println("Starting up Telegram bot...")
+	tg.verbose.LogInfo("Starting up Telegram bot...")
 	var err error
 	tg.api, err = tgbotapi.NewBotAPI(tg.Settings.Token)
 	if err != nil {
-		Error.Println("Failed to connect to Telegram")
+		tg.verbose.LogError(err)
+		// tg.verbose.LogError("Failed to connect to Telegram")
 		errChan <- err
 	}
 	tg.sendToIrc = sendMessage
@@ -65,7 +56,7 @@ func (tg *Client) StartBot(errChan chan<- error, sendMessage func(string)) {
 	updates, err := tg.api.GetUpdatesChan(u)
 	if err != nil {
 		errChan <- err
-		Error.Println(err)
+		tg.verbose.LogError(err)
 	}
 
 	// TODO: Move these lines into the updateHandler when available

--- a/internal/handlers/telegram/telegram.go
+++ b/internal/handlers/telegram/telegram.go
@@ -25,9 +25,7 @@ Creating variables for logging
 var (
 	logFlags    = log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
 	Info        = log.New(os.Stdout, "INFO: ", logFlags)
-	Warning     = log.New(os.Stdout, "WARNING: ", logFlags)
 	Error       = log.New(os.Stderr, "ERROR: ", logFlags)
-	Ignored     = log.New(ioutil.Discard, "ERROR: ", logFlags)
 )
 
 /*
@@ -67,6 +65,7 @@ func (tg *Client) StartBot(errChan chan<- error, sendMessage func(string)) {
 	updates, err := tg.api.GetUpdatesChan(u)
 	if err != nil {
 		errChan <- err
+		Error.Println(err)
 	}
 
 	// TODO: Move these lines into the updateHandler when available

--- a/internal/handlers/telegram/telegram.go
+++ b/internal/handlers/telegram/telegram.go
@@ -14,14 +14,14 @@ the TelegramSettings needed to run the bot
 type Client struct {
 	api       *tgbotapi.BotAPI
 	Settings  internal.TelegramSettings
-	verbose   internal.Debug
+	verbose   internal.DebugLogger
 	sendToIrc func(string)
 }
 
 /*
 NewClient creates a new Telegram bot client
 */
-func NewClient(settings internal.TelegramSettings, tgapi *tgbotapi.BotAPI, debug internal.Debug) *Client {
+func NewClient(settings internal.TelegramSettings, tgapi *tgbotapi.BotAPI, debug internal.DebugLogger) *Client {
 	debug.LogInfo("Creating new Telegram bot client...")
 	return &Client{api: tgapi, Settings: settings, verbose: debug}
 }

--- a/internal/handlers/telegram/telegram.go
+++ b/internal/handlers/telegram/telegram.go
@@ -2,7 +2,8 @@
 package telegram
 
 import (
-	"fmt"
+	"log"
+	"io/ioutil"
 
 	tgbotapi "github.com/go-telegram-bot-api/telegram-bot-api"
 	"github.com/ritlug/teleirc/internal"
@@ -19,10 +20,21 @@ type Client struct {
 }
 
 /*
+Creating variables for logging
+*/
+var (
+	logFlags    = log.Ldate | log.Ltime | log.Lmicroseconds | log.Llongfile
+	Info        = log.New(os.Stdout, "INFO: ", logFlags)
+	Warning     = log.New(os.Stdout, "WARNING: ", logFlags)
+	Error       = log.New(os.Stderr, "ERROR: ", logFlags)
+	Ignored     = log.New(ioutil.Discard, "ERROR: ", logFlags)
+)
+
+/*
 NewClient creates a new Telegram bot client
 */
 func NewClient(settings internal.TelegramSettings, tgapi *tgbotapi.BotAPI) *Client {
-	fmt.Println("Creating new Telegram bot client...")
+	Info.Println("Creating new Telegram bot client...")
 	return &Client{api: tgapi, Settings: settings}
 }
 
@@ -40,11 +52,11 @@ StartBot adds necessary handlers to the client and then connects,
 returning any errors that occur
 */
 func (tg *Client) StartBot(errChan chan<- error, sendMessage func(string)) {
-	fmt.Println("Starting up Telegram bot...")
+	Info.Println("Starting up Telegram bot...")
 	var err error
 	tg.api, err = tgbotapi.NewBotAPI(tg.Settings.Token)
 	if err != nil {
-		fmt.Println("Failed to connect to Telegram")
+		Error.Println("Failed to connect to Telegram")
 		errChan <- err
 	}
 	tg.sendToIrc = sendMessage

--- a/internal/handlers/telegram/telegram_test.go
+++ b/internal/handlers/telegram/telegram_test.go
@@ -22,8 +22,11 @@ func TestNewClientBasic(t *testing.T) {
 		ShowKickMessage:     false,
 		MaxMessagePerMinute: 0,
 	}
+	logger := internal.Debug{
+		DebugLevel:  false,
+	}
 	var tgapi *tgbotapi.BotAPI
-	client := NewClient(tgRequiredSettings, tgapi)
+	client := NewClient(tgRequiredSettings, tgapi, logger)
 	assert.Equal(t, client.Settings, tgExpectedSettings, "Basic client settings should be properly set")
 }
 
@@ -46,8 +49,11 @@ func TestNewClientFull(t *testing.T) {
 		ShowKickMessage:     false,
 		MaxMessagePerMinute: 0,
 	}
+	logger := internal.Debug{
+		DebugLevel:  false,
+	}
 	var tgapi *tgbotapi.BotAPI
-	client := NewClient(tgSettings, tgapi)
+	client := NewClient(tgSettings, tgapi, logger)
 	assert.Equal(t, client.Settings, tgSettings, "All client settings should be properly set")
 	assert.NotEqual(t, client.Settings, tgDefaultSettings, "tgSettings should override defaults")
 }

